docs/src/understand/api-client-perspective/swagger.md

```diff
@@ -12,9 +19,33 @@ looking for, double-check the old.
 
 ## New docs
 
-These docs show swagger 2.0:
-
-[new staging swagger page](https://staging-nginz-https.zinfra.io/api/swagger-ui/)
+The Swagger documentation for public endpoints depends on the API version.
+Though, the version prefix in URL paths works for internal (private) endpoints,
+too, they are not versioned.
+
+These pages show [Swagger / OpenAPI 2.0](https://swagger.io/specification/v2/)
+docs:
+
+- Version `v0`:
+    - [new staging swagger page - **public**
+    endpoints](https://staging-nginz-https.zinfra.io/api/swagger-ui/)
+- Version `v1`:
+    - [new staging swagger page - **public**
+    endpoints](https://staging-nginz-https.zinfra.io/v1/api/swagger-ui/)
+- Version `v2`:
+    - [new staging swagger page - **public**
+    endpoints](https://staging-nginz-https.zinfra.io/v2/api/swagger-ui/)
+- Version `v3` (current *development* version):
+    - [new staging swagger page - **public**
+    endpoints](https://staging-nginz-https.zinfra.io/v3/api/swagger-ui/)
+- Unversioned
+    - [new staging swagger page - **internal** (private)
+    endpoints](https://staging-nginz-https.zinfra.io/api-internal/swagger-ui/)
```

Comment:
I guess it's silly to generate this list, but maybe we can write it up so that the reader can construct the correct URL themselves?  (put an `N` in there and ask the user to replace it, and find the latest version number under `/api-versions`?)

That's slightly more awkward, but also a lot more future-proof then this.

Also, I think internal API is just as versions as external (or should be); see above.

Reply:
wait -- does skippping the version path segment give you V0 or the most recent version?  V0 would be consistent with all the other paths, most recent would be most useful, "error - please pick a version number" would be most robust and least misleading.  i'm in favor of either useful or robust.

Reply:
Paolo (not using GitHub mention by intention) explained this detail in https://docs.wire.com/developer/developer/api-versioning.html?highlight=versioning#making-requests-to-a-particular-version :

> To support clients that have not yet implemented versioning, backends that support version 0 will also accept unversioned requests, which will automatically be rewritten to /v0/ requests.

I agree with you proposal regarding explaining the URL pattern vs. a prepared list. I'll adjust the docs accordingly.

Reply:
yes, i'm aware of how "no version segment" works for the rest of the api, i just thought since clients won't have to call swagger code in the wild we could get away with giving /api/swagger-ui and /api/swagger-docs a different semantics.

anyway, the important thing is that the docs explain it correctly and for all future versions.